bundles/org.openhab.binding.huesync/README.md

```diff
@@ -0,0 +1,288 @@
+# HueSync Binding
+
+<!-- markdownlint-disable MD033 -->
+This binding integrates the [Play HDMI Sync Box](https://www.philips-hue.com/en-us/p/hue-play-hdmi-sync-box-/046677555221) into openHAB.
+The integration happens directly through the Hue [HDMI Sync Box API](https://developers.meethue.com/develop/hue-entertainment/hue-hdmi-sync-box-api/).
+
+- [HueSync Binding](#huesync-binding)
+  - [Discovery](#discovery)
+  - [Thing Configuration](#thing-configuration)
+  - [Channels](#channels)
+    - [Channels - Groups](#channels---groups)
+    - [Channels - device-firmware](#channels---device-firmware)
+    - [Channels - device-hdmi-connection-\[in|out\]](#channels---device-hdmi-connection-inout)
+      - [device-commands](#device-commands)
+  - [Item Configuration](#item-configuration)
+    - [Items - Groups](#items---groups)
+    - [Items - Remote Control](#items---remote-control)
+    - [Items - Firmware](#items---firmware)
+
+## Discovery
+
+The binding is using [mDNS](https://en.wikipedia.org/wiki/Multicast_DNS) to discover HDMI Sync devices in the local network.
+The LED on the Sync Box must be white or red.
+This indicates that the device is connected to the Network.
+If the LED is blinking blue, you need to setup the device using the official [Hue Sync App](https://www.philips-hue.com/en-in/explore-hue/propositions/entertainment/hue-sync).
+
+If the device is not discovered you can check if it is properly configured and discoverable in the network:
+
+<details>
+  <summary>Linux (Ubuntu based distributions)</summary>
+
+```bash
+$ avahi-browse --resolve _huesync._tcp 
++ wlp0s20f3 IPv4 HueSyncBox-XXXXXXXXXXX                       _huesync._tcp        local
+= wlp0s20f3 IPv4 HueSyncBox-XXXXXXXXXXX                       _huesync._tcp        local
+   hostname = [XXXXXXXXXXX.local]
+   address = [192.168.0.12]
+   port = [443]
+   txt = ["name=Sync Box" "devicetype=HSB1" "uniqueid=XXXXXXXXXXX" "path=/api"]
+```
+
+</details>
+
+## Thing Configuration
+
+### Registration
+
+To communicate with the sync box, you need to couple the thing with the hardware (registration). The thing will start this process automatically. To complete the registration you just press the "coupling" button on the sync box for 3 seconds.:
```

Comment:
Thanks! The logging is still no my todo -... but I'll have a look at the guidelines. The markdown mistake I make every time. I wonder if the linter could mark them as error as well - so that you and other reviewers do not need to waste time on such trivialities. Same goes for naming conventions - as every project I'm working on uses different conventions and sometimes I do not work on one of them for a while it is easy to do it wrong (and if I read the coventions every time again, the timeslot is already over ...).

The CI/CD and linter of the project already has a very high quality. I'm not familiar with the stack used - but maybe the investment in some more linter rules would increase productivity and review overhead. Do you know who is the maintainer of the pipeline? Then I could ask if that's easy to achieve - or a bigger project.

Something else the README: I had a look at the Hue binding README as a reference and noticed, that it used some images for visualization. We removed them from the README; but honestly if possible I would like to re-introduce them - as it seems other bindings do so as well and I realy think the documentation looks better without them. Would it be acceptable to re-introduce them?

https://www.openhab.org/addons/bindings/hue/#philips-hue-binding

Reply:
Confectrician did a lot of work on the documentation/linter. I use vscode and it markes all the issue by default (atleast i'm not aware that i added an extension). Maybe the static code analysis can be extended to cehck the thing-structure.xml files for naming convention. If you can make the life easier for contributers and maintainers, your are very welcome!

Regarding the images i prefer that you leave them out.

Reply:
Now I understand what you refered to with the work of @Confectrician - unfortunately I was not aware of the detailed static code analysis report created by the [CI](https://ci.openhab.org/job/PR-openHAB-Addons/24272/Static_20Code_20Analysis_20Report/).

I'll try to include this information also in the docu. I assumed I used the correct .xml rule set as described by the docu to catch those locally already - but did something wrong; or the pipeline has some more rules or tooling.

➡️ I'll address the warnings and check my config again. If I do not find the mistake I'll contact Confectrician for help to clarify the docu and if it is possible to perform those checks directly on the dev machine.

April_Wexler (Kai) started with the work on the README - let's see - maybe it will make it into 4.3 after all.

Reply:
@Confectrician 

Hi Jerome,

while checking the CI build I noticed that I still have some policy violations in my binding: 

https://ci.openhab.org/job/PR-openHAB-Addons/24272/Static_20Code_20Analysis_20Report/

To avoid that I miss such things I would like to configure my IDE (Visual Studio Code), that I also get this violations/warnings locally.

Those are the settings I use  (relevant parts only):
```json
"java.codeGeneration.generateComments": true,
    "java.compile.nullAnalysis.mode": "automatic",
    "java.configuration.checkProjectSettingsExclusions": false,
    "java.configuration.updateBuildConfiguration": "interactive",
    "java.debug.settings.console": "integratedTerminal",
    "java.debug.settings.hotCodeReplace": "never",
    "java.format.comments.enabled": true,
    "java.format.enabled": true,
    "java.format.onType.enabled": true,
    "java.format.settings.profile": "openHAB",
    "java.format.settings.url": "https://raw.githubusercontent.com/openhab/static-code-analysis/main/codestyle/src/main/resources/openhab_codestyle.xml",
    "java.help.collectErrorLog": true,
    "java.jdt.ls.java.home": "/usr/lib/jvm/java-17-openjdk-amd64",
    "java.project.resourceFilters": [
        "node_modules",
        "\\.git",
        "\\.vscode"
    ],
    "java.saveActions.organizeImports": true,
    "jdk.serverVmOptions": [],
    "maven.pomfile.autoUpdateEffectivePOM": true,
    "maven.terminal.useJavaHome": true,
    "maven.view": "hierarchical",
```

I will also include the information on how this configuration can be done in the documentation (https://github.com/openhab/openhab-docs/pull/2400) ...

thank you for the excellent work done on the pipeline; contributing to this project is a great way to learn from others while doing something fun and useful 🙂.

with kind regards,
Patrik